gluten-substrait/src/main/scala/org/apache/gluten/execution/BatchScanExecTransformer.scala

```diff
@@ -96,18 +96,22 @@ abstract class BatchScanExecTransformerBase(
   // class. Otherwise, we will encounter an issue where makeCopy cannot find a constructor
   // with the corresponding number of parameters.
   // The workaround is to add a mutable list to pass in pushdownFilters.
-  protected var pushdownFilters: Option[Seq[Expression]] = None
+  protected var pushdownFilters: Seq[Expression] = scan match {
+    case fileScan: FileScan =>
+      fileScan.dataFilters.filter {
+        expr =>
+          ExpressionConverter.canReplaceWithExpressionTransformer(
+            ExpressionConverter.replaceAttributeReference(expr),
+            output)
+      }
+    case _ => Seq.empty
```

Comment:
Perhaps add some warnings for easy debugging.

Reply:
I don’t quite understand what the warning is here. Can you explain it? Thank you. @rui-mo

Reply:
Previously we throw an exception if it is not file scan. Do we need to log the information for unsupported cases?
https://github.com/apache/incubator-gluten/pull/8082/files#diff-0637bf279d5bc9ae2791c4e8919675e1674426634f187dfc55596f7c1eb4c5afL109